### DISCUSSION TITLE
Oneplus 6 OEM unlocking option missing

### DISCUSSION THREAD

**Discussion by kinau12** [Divested-Mobile/DivestOS-Build#325] [General] [answered]
I'm running divestos 20-20240712 on my oneplus 6, and have relocked the bootloader. After installation, I enabled OEM unlocking in the developer (as suggested), but now this option seems to be gone entirely from the settings. I've tried to reset the device but no avail. I also tried fastboot oem unlock, but it gave me FAILED (remote: 'Flashing Unlock is not allowed, thus indicating that OEM unlocking is disabled. I really would need the OEM unlock option back. Any help would be appreciated.

**Comment by SkewedZeppelin** [comment, marked as answer]
I've had a few reports of this, but the toggle is supposed to always be force-enabled/visible.
https://github.com/Divested-Mobile/DivestOS-Build/blob/master/Scripts/Common/Enable_Verity.sh#L240

It shows up just fine on my OnePlus 6T/fajita as well, so I haven't been able to debug it.

However this device has EDL enabled so you can always restore it back to stock even if locked using msmdownloadtool.
There are guides on XDA to do this, eg. https://xdaforums.com/t/op6-latest-10-3-8-collection-of-unbrick-tools.3914109/

**Reply by kinau12** [reply]
Thank you for pointing me to right direction. I ended up using this guide: https://community.oneplus.com/thread/1541327, which worked perfectly; the main hurdle was getting a Windows computer in my hands ;-)

**Reply by SkewedZeppelin** [reply]
https://github.com/bkerler/edl works without Windows, but is a bit more manual

**Comment by kinau12** [comment]
Cool, thank you for letting me know!
